src/io/hdf5_idas.py

```diff
@@ -0,0 +1,130 @@
+    'gauge_length': 'GaugeLength',
+    'norm_offset': 'NormOffset',
+    'source_mode': 'SourceMode',
+    'time_decimation': 'TimeDecimation',
+    'zero_offset': 'ZeroOffset',
+    'p_parameter': 'P',
+    'p_coefficients': 'P_Coefficients',
+    #'idas_version': 'iDASVersion',
+    'precice_sampling_freq': 'PreciseSamplingFrequency',
+    'receiver_gain': 'ReceiverGain',
+    #'continuous_mode': 'Continuous Mode',
+    'geo_lat': 'Latitude',
+    'geo_lon': 'Longitude',
+    'geo_elevation': 'Altitude',
+
+    'channel': None,
+    'unit': 'RawDataUnit' 
+}
+
+
+def get_meta(h5file):
+    """
+    Get metadata from HDF5 file using (almost) the same fields as for the TDMS files.
+    
+    Parameters
+    ----------
+    h5file : HDF5 file object
+        The file to extract metadata from
+    Returns
+    -------
+        Dictionary containing the metadata 
+
+    """
+
+    key_list = list(META_KEYS.keys())
+    val_list = list(META_KEYS.values())
+
+    meta = dict()
+    # the following is based on the PRODML format
+    for field in ['Acquisition',
+                  'Acquisition/Custom/AdvancedUserSettings',
+                  'Acquisition/Custom/SystemSettings',
+                  'Acquisition/Custom/SystemInformation/GPS',
+                  'Acquisition/Custom/UserSettings',
+                  'Acquisition/Raw[0]']:
+        try:
+            field_keys = h5file[field].attrs.keys()
+            for val in val_list:
+                if val in field_keys:
+                    meta[key_list[val_list.index(val)]] = f[field].attrs[val]
+        except:
+            pass
+                
+    # some type conversions
+    for val in ['p_coefficients', 'receiver_gain', 'source_mode', 'unit']:
+        if val in meta.keys():
+            meta[val] = meta[val].decode("utf-8")
+    for val in ['decimation_filter', 'normalization']:
+        if val in meta.keys():
+            meta[val] = bool(meta[val])
+    
+    return meta
+
+
+def iload(filename, load_data=True):
+
+    # prevent hard dependency on h5py
+    try:
+        import h5py
+    except Exception as e:
+        print(e)
+        raise ImportError("Please install 'h5py' to proceed, e.g. by running 'pip install h5py'")
+
+
+    with h5py.File(filename, "r") as f:
+        # get the meta data
+        meta = get_meta(f)
+        # get the actual time series if load_data
+        data = f['Acquisition/Raw[0]/RawData'][:].copy() if load_data else None
+        # get the sampling rate, starttime, number of samples in space and time
+        deltat = 1. / f['Acquisition/Raw[0]'].attrs['OutputDataRate']
+        tmin = datetime.strptime(
+            f['Acquisition/Raw[0]/RawData'].attrs['PartStartTime'].decode('ascii'),
+            '%Y-%m-%dT%H:%M:%S.%f+00:00').timestamp()
+        nchan = f['Acquisition/Raw[0]'].attrs['NumberOfLoci']
+        try:
+            # usually this should be here?
+            nsamp = f['Acquisition/Raw[0]/RawDataTime'].attrs['Count'] 
+        except:
```

Comment:
Please use an explicit Exception. `KeyError`?

Reply:
I removed this try-except statement and only kept line 103, which should work for files with complete header information.